### PR TITLE
Enable Forward declaration for types used in unique ptrs in USpatialNetDriver

### PR DESCRIPTION
**Pull request by ImprobableNic**
#### Description

We can actually forward declare types referenced through UniquePtr. It was a bit tricky when done with UObject because there is an obscure contructor that has to be manually defined.
I cleaned up USpatialNetDriver to reduce the amount of header pulled in, it should be done as often as possible.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriver.h [hunk 1]
```diff
@@ -54,6 +45,8 @@ class USpatialStaticComponentView;
 class USpatialWorkerConnection;
 class USpatialWorkerFlags;
 
+DECLARE_DELEGATE(PostWorldWipeDelegate);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriver.h:48]: What's this?

Comment (ImprobableNic) [reply SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriver.h:48]: Something that's defined in SnapshotManager, and that the NetDriver needs. I just reported it as a "forward declaration" of sorts.

File: SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriver.h [hunk 2]
```diff
@@ -85,6 +90,7 @@ class SPATIALGDK_API USpatialNetDriver : public UIpNetDriver
 
 public:
 	USpatialNetDriver(const FObjectInitializer& ObjectInitializer);
+	USpatialNetDriver(FVTableHelper& Helper);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriver.h:93]: Hmm Unreal has a big `Don't use` around this functionality -
```
#define DECLARE_VTABLE_PTR_HELPER_CTOR(API, TClass) \
	/** DO NOT USE. This constructor is for internal usage only for hot-reload purposes. */ \
```

Comment (ImprobableNic) [reply SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriver.h:93]: I'ts more of a "Do not call". I expect this macro to be used in generated code.

**Review by m-samiec — APPROVED** [review]
Nice!

**Pull request merged**